### PR TITLE
move etcd client to gost

### PR DESCRIPTION
**Pull request by wenxuwan**
**What this PR does**:

Move etcd client to gost

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -54,6 +54,9 @@ require (
 
 replace (
 	github.com/coreos/bbolt => go.etcd.io/bbolt v1.3.4
+	github.com/dubbogo/gost => github.com/wenxuwan/gost v1.9.4-0.20210309022905-7993b30b1a7d
```
Comment (zouyx) [inline go.mod:57]: is it right replacment?

File: remoting/etcdv3/listener_test.go [hunk 2]
```diff
@@ -18,17 +18,24 @@
...
+	"github.com/coreos/etcd/embed"
+	gxetcd "github.com/dubbogo/gost/database/kv/etcd/v3"
 	"github.com/stretchr/testify/assert"
 )
 
 import (
 	"github.com/apache/dubbo-go/remoting"
 )
 
+const defaultEtcdV3WorkDir = "/tmp/default-dubbo-go-remote.etcd"
```
Comment (zouyx) [inline remoting/etcdv3/listener_test.go:37]: will it cause problem in docker env? like access deny , can not open file etc?

File: go.mod (hunk 1 above)
Comment (wenxuwan) [reply go.mod:57]: No, just for test, when this pr https://github.com/dubbogo/gost/pull/47 merged, will replace it.

File: remoting/etcdv3/listener_test.go (hunk 2 above)
Comment (wenxuwan) [reply remoting/etcdv3/listener_test.go:37]: May be is ok, i just use the old dir in database/kv/etcd/v3/client_test.go, so should be ok.

**Comment by AlexStocks** [comment]
pls move the target branch to 3.0.

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/apache/dubbo-go
 
-go 1.13
+go 1.15
```
Comment (zouyx) [inline go.mod:3]: 1.15 at least?

Comment (wenxuwan) [reply go.mod]: 1.13 should be ok ,change back

Comment (wenxuwan) [reply go.mod:3]: zhixin use 1.15, so keep change here

**Review by AlexStocks — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -36,15 +36,15 @@ require (
...
+	github.com/prometheus/client_golang v1.9.0
 	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b
 	github.com/stretchr/objx v0.2.0 // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/zouyx/agollo/v3 v3.4.5
 	go.uber.org/atomic v1.7.0
 	go.uber.org/zap v1.16.0
 	golang.org/x/sys v0.0.0-20201223074533-0d417f636930 // indirect
-	google.golang.org/grpc v1.26.0
+	google.golang.org/grpc v1.33.1
```
Comment (LaurenceLiZhixin) [inline go.mod:47]: Pls use grpc 1.26.0

Comment (wenxuwan) [reply go.mod:47]: i replace it on 60 line

**Review by ztelur — APPROVED** [review]

**Review by zhaoyunxing92 — APPROVED** [review]
approved these changes

**Pull request merged**